Neptune.WebMvc/Controllers/OnlandVisualTrashAssessmentAreaController.cs

```diff
@@ -327,6 +338,215 @@ public async Task<ActionResult> NewAssessment([FromRoute]
+                        }
+                    }
+                };
+                var geoJson = await _gdalApiService.Ogr2OgrGdbToGeoJson(apiRequest);
+                var ovtaAreaStagings = await GeoJsonSerializer.DeserializeFromFeatureCollectionWithCCWCheck<OnlandVisualTrashAssessmentAreaStaging>(geoJson,
+                    GeoJsonSerializer.DefaultSerializerOptions, Proj4NetHelper.NAD_83_HARN_CA_ZONE_VI_SRID);
+                // todo: Run MakeValid "update dbo.DelineationStaging set Geometry = Geometry.MakeValid() where Geometry.STIsValid() = 0";
+
+                var validOVTAAreaStagings = ovtaAreaStagings.Where(x => x.Geometry is { IsValid: true, Area: > 0 }).ToList();
+                if (validOVTAAreaStagings.Any())
+                {
+                    await _dbContext.OnlandVisualTrashAssessmentAreaStagings.Where(x => x.UploadedByPersonID == CurrentPerson.PersonID).ExecuteDeleteAsync();
+                    _dbContext.OnlandVisualTrashAssessmentAreaStagings.AddRange(validOVTAAreaStagings);
+                    await _dbContext.SaveChangesAsync();
+                }
+            }
+            catch (Exception e)
+            {
+                if (e.Message.Contains("Unrecognized field name",
+                        StringComparison.InvariantCultureIgnoreCase))
+                {
+                    ModelState.AddModelError("",
+                        "The columns in the uploaded file did not match the OVTA area schema. The file is invalid and cannot be uploaded.");
+                }
+                else
+                {
+                    ModelState.AddModelError("",
+                        $"There was a problem processing the Feature Class \"{featureClassName}\". The file may be corrupted or invalid.");
+                }
+                return ViewUpdateOVTAAreaGeometryErrors(viewModel);
+            }
+
+
+            return RedirectToAction(new SitkaRoute<OnlandVisualTrashAssessmentAreaController>(_linkGenerator, x => x.ApproveOVTAAreaGisUpload()));
+        }
+
+        [HttpGet]
+        [JurisdictionManageFeature]
+        public ActionResult ApproveOVTAAreaGisUpload()
+        {
+            var viewModel = new ApproveOVTAAreaGisUploadViewModel();
+            return ViewApproveOVTAAreaGisUpload(viewModel);
+        }
+
+        [HttpPost]
+        [JurisdictionManageFeature]
+        public async Task<ActionResult> ApproveOVTAAreaGisUpload(ApproveOVTAAreaGisUploadViewModel viewModel)
+        {
+            if (!ModelState.IsValid)
+            {
+                return ViewBulkUploadOTVAAreas(new BulkUploadOVTAAreasViewModel());
+            }
+
+            var onlandVisualTrashAssessmentAreaStagings = _dbContext.OnlandVisualTrashAssessmentAreaStagings.Where(x => x.UploadedByPersonID == CurrentPerson.PersonID).ToList();
+
+            //// Will break if there are multiple batches of staged uploads, which is precisely what we want to happen. 

```

Comment:
I don't think this comment is accurate; what the comment should say is that we should only be doing one jurisdiction at a time